modules/cli/src/main/scala/scala/cli/commands/SharedOptions.scala

```diff
@@ -101,7 +102,9 @@ final case class SharedOptions(
   @Hidden
     forbid: List[String] = Nil,
   @Recurse
-  helpGroups: HelpGroupOptions = HelpGroupOptions()
+    helpGroups: HelpGroupOptions = HelpGroupOptions(),
+  @Recurse
+    launcherOptions: LauncherOptions = LauncherOptions()
```

Comment:
Maybe we shouldn't put those here, but rather manually parse those, requiring those options to appear first. Something along those lines in `ScalaCli.main` should work:
```scala
val remainingArgs = LauncherOptions.parser.stopAtFirstUnrecognized.parse(args) match {
  case Left(error) =>
    ??? // print error and exit
  case Right((launcherOpts, args0)) =>
    launcherOpts.scalaCliVersion.map(_.trim).filter(_.nonEmpty) match {
      case Some(ver) =>
        LauncherCli.runAndExit(ver, args0)
      case None =>
        args0
    }
}
```